Add hiv resistance detection (#559)
nf-core/viralrecon: docs/usage.md

- `--genome 'NC_001802.1'`: nextclade refeence, based on NC_001802.1 (HXB2 genome, K03455), the reference used by Nextclade for HIV-1 analysis.

#### **HIV-specific parameters**

The following parameters have been added to handle **HIV resistance detection** and related tools.

- General HIV resistance options:
  - **`perform_hiv_resistance`**: Whether to perform HIV resistance analysis or not.
- Options related with files required by `sierra-local` software for resistance detecton in HIV. If not provided, the files included with the software will be used.
  - **`hivdb_xml`**: Path to the HIVdb ASI2 XML file. Updated files can be found [here](https://github.com/hivdb/hivfacts/tree/main/data/algorithms), where the different algorithm versions are stored. The one included in the software is the one with `HIVDB_9.8.xml` name.
  - **`apobec_drm`**: Path to the JSON HIVdb APOBEC DRM file. Updated files can be found [here](https://github.com/hivdb/hivfacts/tree/main/data/apobecs) with the `apobec_drms.json` name.

>   - **`apobec_csv`**: Path to the CSV APOBEC file.Updated files can be found [here](https://github.com/hivdb/hivfacts/tree/main/data/apobecs) with the `apobecs.csv` name.

Reviewer: ```suggestion
  - **`apobec_csv`**: Path to the CSV APOBEC file. Updated files can be found [here](https://github.com/hivdb/hivfacts/tree/main/data/apobecs) with the `apobecs.csv` name.
```

--fasta <path_to_reference.fasta>
--gff   <path_to_annotation.gff>
```

## HIV

`nf-core/viralrecon` has been optimized to support **HIV genome reconstruction and resistance detection**.

This implementation takes inspiration from the parameterization used in the [**HIVdb** pipeline at Stanford University](https://hivdb.stanford.edu/hivdb/by-reads/).

The goal is to adapt the viralrecon framework to ensure accurate minor variant calling, reliable consensus sequence generation, and compatibility with resistance analysis tools for HIV.

### HIV profile and recomended params

The HIV profile (`-profile hiv`) introduces specific adjustments compared to a standard `viralrecon` run, mainly in the variant calling and consensus generation steps, to align with the requirements of HIV resistance interpretation pipelines.

> ⚠️ By deffault, de novo assembly is deactivated for HIV as the resistance detection is only performed for the consensus genome through mapping approach, which has been already benchmarked agains Standford's [HIVdb](https://hivdb.stanford.edu/hivdb/by-reads/) results.

Reviewer: ```suggestion
⚠️ By default, de novo assembly is deactivated for HIV as the resistance detection is only performed for the consensus genome through mapping approach, which has been already benchmarked agains Stanford's [HIVdb](https://hivdb.stanford.edu/hivdb/by-reads/) results.
```

It allows laboratories to generate **HIV-1 drug resistance predictions locally**, without the need to transmit patient data over a network. This provides full control over **data provenance**, **security**, and **regulatory compliance**.

> _Sierra-local computes drug resistance predictions directly from consensus HIV-1 sequences, producing JSON-formatted output that includes key resistance-associated mutations and drug susceptibility scores._

Within the HIV module of `nf-core/viralrecon`, sierra-local is executed using the consensus .FASTA file generated for each sample. The output is a `.json` file containing the main resistance data.

However, sierra-local operates at the **codon level**, and its JSON output does not include **codon-level frequencies** or **codon depth information**.  
These additional metrics are crucial for resistance reporting and neeed consistency with other viralrecon outputs, which are based on nucleotide-level variant calls.

#### **2. Codon-level frequency analysis with codfreq**

To obtain codon frequency information, `nf-core/viralrecon` integrates a modified version of [**codfreq**](https://github.com/hivdb/codfreq/) tool.

> The modified cersion of codfreq produces a codon frequency table in the _CodFreq_ format, which contains seven columns:

Reviewer: ```suggestion
The modified version of codfreq produces a codon frequency table in the _CodFreq_ format, which contains seven columns:
```

| `aa_codon`            | If the codon is multiple of 3, the corresponding aminoacid sequence |

This format enables codon-level analysis of intra-sample diversity.

However, codfreq requires a **custom JSON file** that describes the HIV gene structure (PR, RT, and IN). To generate this JSON dynamically according to the user’s selected reference genome, several preprocessing steps are performed.

#### **3. Generation of codfreq annotation JSON**

The generation of the required JSON file for codfreq involves the following steps:

1. **Conversion of reference annotations**: The original codfreq HIV profile (in JSON format) has been converted into a `.fasta` and `.gff` file containing the necessary annotations for PR, RT, and IN genes. These files are included within `nf-core/viralrecon` assets.

2. **Reference genome alignment with Liftoff**: Only performed when the provided genome is not the one from codfreq. Using the provided reference genome and the `.fasta` and `.gff` from the step 1, annotation files are processed with **[Liftoff](https://github.com/agshumate/Liftoff)** to transfer the HIV gene annotations to the user’s specific reference genome.

> 3. **Custom JSON generation**: The annotated `.gff` file and the reference `.fasta` are used as input for a **custom Python script** within `nf-core/viralrecon`. This script produces the JSON file required by `codfreq`, ensuring accurate gene coordinates and consistency with the reference genome.

Reviewer: ```suggestion
3. **Custom JSON generation**: The annotated `.gff` file and the reference `.fasta` are used as input for a **custom Python script** `bin/gff2json.py` within `nf-core/viralrecon`. This script produces the JSON file required by `codfreq`, ensuring accurate gene coordinates and consistency with the reference genome.
```
